iib/web/models.py

```diff
@@ -1049,6 +1071,99 @@ def get_mutable_keys(self):
         return rv
 
 
+class RequestMergeIndexImage(Request, RequestIndexImageMixin):
+    """An "merge-index-image" build request."""
+
+    __tablename__ = 'request_merge_index_image'
+
+    id = db.Column(db.Integer, db.ForeignKey('request.id'), autoincrement=False, primary_key=True)
+    source_from_index_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    source_from_index = db.relationship('Image', foreign_keys=[source_from_index_id], uselist=False)
+    target_from_index_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    target_from_index = db.relationship('Image', foreign_keys=[target_from_index_id], uselist=False)
+    deprecation_list = db.relationship('Image', secondary=BundleDeprecation.__table__)
+
+    __mapper_args__ = {
+        'polymorphic_identity': RequestTypeMapping.__members__['merge_index_image'].value,
+    }
+
+    @classmethod
+    def from_json(cls, kwargs, batch=None):
+        """
+        Handle JSON requests for the Add API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        bundles = request_kwargs.pop('deprecation_list', [])
+        if not isinstance(bundles, list) or any(
+            not item or not isinstance(item, str) or '@' not in item for item in bundles
```

Comment:
Why do you need the bundles to be specified with their hash? Doesn't IIB already resolve them to their hases? You could reuse that method

Reply:
I think I saw the requirement for the bundles to be specified by a digest in the design doc. "It’ll be a yaml file with package name as the key and the bundle pull_spec list as value specified with digest. " I guess the data validation doesn't have to be done on IIB's side, but should instead be handled by the script. Should I changed it to accept tags as well?

Reply:
I think all you need to check here is that bundles is an array of strings. Let the bundle resolution in the code take care of valid/invalid bundles.

Reply:
The reason why I added it was to check if each bundle is specified via digest, not to verify bundle validity.

I can still remove it if you'd like.